handler/handler.go

```diff
@@ -76,5 +76,6 @@ func SetResponseHeader(name, value string) {
 func SendResponse(statusCode uint32, body []byte) {
```

Comment:
I like how this is "an alternative to Next" so kept this as is for now. We'd probably need to expose the setstatus / body methods separately for advanced usage, can do that now if it makes sense

Reply:
I like how we are thinking about the API vs the ABI, as the latter might get trickier over time. The only thought I have is if "SendResponse" is misleading because it won't necessarily send it (likely afterwards). so only naming and that can be handled later also.

Reply:
Can't come up with any clear advantages of this or alternatives that come to mind so will leave for now and think some more